### PR TITLE
New production-ready environment

### LINKED ISSUE #51: Add the ability to deploy a production-ready cluster
Hi team,

Currently, we provide a way to deploy a specific environment with several customized use cases installed on different agents. The main goal of this task is to implement the ability to deploy a production-ready cluster that includes:

- [x] Wazuh cluster of 2 manager nodes.
- [x] Secured Elasticsearch cluster of 3 nodes (XPack security).
- [x] Secured Kibana interface.
- [x] Agents can be registered and report to the cluster using a load balancer.
- [x] Optionally, include Route53 records.
- [x] Related documentation

The template should support as many parameters as possible in order to gain flexibility. Also, this environment should be started with the default configuration.

Regards

### PR DESCRIPTION
**Pull request by Zenidd**
Hi team,

this PR adds a new production-ready environment useful for the community users. 
this PR closes #51

Greetings, 
JP

### CODE REVIEW

File: demo/elastic/wazuh_cf_kibana.sh [hunk 1]
```diff
@@ -297,6 +297,18 @@ install_plugin(){
 
 }
 
+
+optimize_kibana(){
+  systemctl stop kibana
+  echo "Optimizing app" >> /tmp/deploy.log
+  cd /usr/share/kibana
+  NODE_OPTIONS="--max-old-space-size=2048" ./bin/kibana --optimize --allow-root
```
Comment (jm404) [inline demo/elastic/wazuh_cf_kibana.sh:305]: Why doing the optimize as root?

File: production/deploy.sh [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+# Checking arguments
+if [ "$STACK_NAME" == "" ] || [ "$BUCKET_NAME" == "" ]; then
+    echo "Missing template path or parameters file."
+    exit
+fi
+
+# Uploading template to S3
+aws s3 cp $TEMPLATE_FILE s3://$BUCKET_NAME
+# Getting the template URL
+URL="https://<S3-bucket-name>.amazonaws.com/wazuh_template.yml"
```
Comment (jm404) [inline production/deploy.sh:32]: This Variable could use `${BUCKET_NAME}` to be rendered dynamically, right?

**Comment by jm404** [comment]
The PR shows some conflicts, please check them

File: production/deploy.sh (hunk 2 above)
Comment (Zenidd) [reply production/deploy.sh]: Done in 3ad99aba96b01b0a6f9e944ba64c25d0c7c3a9c0. Thanks !

File: demo/elastic/wazuh_cf_kibana.sh (hunk 1 above)
Comment (Zenidd) [reply demo/elastic/wazuh_cf_kibana.sh]: Fixed on 4672cd5e65ea9ed2db6de6ed06472afe74eddcca.

**Pull request merged**